conformance/utils/http/http.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+	"sigs.k8s.io/gateway-api/conformance/utils/roundtripper"
+)
+
+// ExpectedResponse defines the response expected for a given request.
+type ExpectedResponse struct {
+	Request    ExpectedRequest
+	StatusCode int
+	Backend    string
+	Namespace  string
+}
+
+// ExpectedRequest can be used as both the request to make and a means to verify
+// that echoserver received the expected request.
+type ExpectedRequest struct {
+	Host    string
+	Method  string
+	Path    string
+	Headers map[string]string
+}
+
+// MakeRequestAndExpectResponse makes a request with the given parameters and
```

Comment:
Is this function actually used anywhere?

Reply:
It is used quite heavily by the tests I pulled out to #994, have pulled out this function to add back with that PR.